### PR TITLE
Adding default email template

### PR DESCRIPTION
**Pull request by Vardane**
| Q                                          | R
| ------------------------------------------ | -------------------------------------------
| Type of contribution ?                     | Enhancement
| Tickets (_issues_) concerned               | #246 

---

##### What have you changed ?
- send_email_confirmation function and send_email_cancellation_emergency in Participation class
- added html templates as well as images in static files

##### How did you change it ?
- added feature allowing using a default email template

##### Is there a special consideration?
- some redundancy remains in html files, which use the same hard-coded css classes, in order to accommodate SendinBlue behavior. 

Verification :
 -  [x] My branch name respect the standard defined in `CONTRIBUTING.md`
 -  [x] All my commits respect the standard defined in `CONTRIBUTING.md`
 -  [x] This Pull-Request fully meets the requirements defined in the issue
 -  [x] I added or modified the attached tests
 -  [x] I added or modified the documentation

### CODE REVIEW

**Comment by Vardane** [comment]
@RignonNoel The test test_send_email_confirmation in test_view_participations returns an error. Message says that 3 emails are sent, while I was expecting (and testing for) one (I assume a single confirmation is sent to user).
I wonder: are 3 emails actually sent every time a new position is created? Or is it because the testing context keeps in memory positions created when other test functions where run in the same series of tests? Or some other reason?

Error message:

Traceback (most recent call last):
  File "/home/vardane/Documents/api_volontaria/API-Volontaria/api_volontaria/apps/volunteer/tests/test_view_participations.py", line 417, in test_send_email_confirmation
    self.assertEqual(len(mail.outbox), 1)
AssertionError: 3 != 1

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "python.linting.pylintEnabled": true,
+    "python.linting.enabled": true
+}
```
Comment (RignonNoel) [inline .vscode/settings.json:4]: `.vscode` -> should be added to `.gitignore`

File: api_volontaria/apps/user/tests/tests_view_users.py [hunk 2]
```diff
@@ -14,6 +14,7 @@
 from ..models import ActionToken
 from ....testClasses import CustomAPITestCase
 
+from api_volontaria.apps.log_management.models import EmailLog
```
Comment (RignonNoel) [inline api_volontaria/apps/user/tests/tests_view_users.py]: You just added this line but it's not used so you can remove it

File: api_volontaria/apps/volunteer/models.py [hunk 3]
```diff
@@ -5,12 +5,15 @@
...
 from django.dispatch import receiver
+from django.template.loader import render_to_string
 from django.utils import timezone
 from django.utils.translation import ugettext_lazy as _
 from django.contrib.auth import get_user_model
 from dry_rest_permissions.generics import authenticated_users
 from api_volontaria.email import EmailAPI
 
+# from config import settings
+
```
Comment (RignonNoel) [inline api_volontaria/apps/volunteer/models.py]: Code should not be commented but removed

File: api_volontaria/apps/volunteer/tests/test_view_participations.py [hunk 4]
```diff
@@ -383,3 +392,306 @@ def test_list_participations_as_admin(self):
...
+        self.client.force_authenticate(user=self.user)
+
+        response = self.client.post(
+            reverse('participation-list'),
+            data_post,
+
+            format='json',
+        )
+
+        # 1. email sent?
```
Comment (RignonNoel) [inline api_volontaria/apps/volunteer/tests/test_view_participations.py:466]: Les petit commentaires comme celui-ci peuvent etre enlever maintenant

File: api_volontaria/apps/volunteer/tests/test_view_participations.py [hunk 5]
```diff
@@ -383,3 +392,306 @@ def test_list_participations_as_admin(self):
...
+        nb_email_sent = len(mail.outbox) - outbox_initial_email_count
+
+        # 2.a and 2.b: proper template?
```
Comment (RignonNoel) [inline api_volontaria/apps/volunteer/tests/test_view_participations.py]: Idem

File: api_volontaria/apps/volunteer/tests/test_view_participations.py [hunk 6]
```diff
@@ -383,3 +392,306 @@ def test_list_participations_as_admin(self):
...
+        self.client.force_authenticate(user=self.user)
+
+        response = self.client.post(
+            reverse('participation-list'),
+            data_post,
+
+            format='json',
+        )
+
+        # 1. email sent?
```
Comment (RignonNoel) [inline api_volontaria/apps/volunteer/tests/test_view_participations.py:543]: Idem

File: api_volontaria/apps/volunteer/tests/test_view_participations.py [hunk 7]
```diff
@@ -383,3 +392,306 @@ def test_list_participations_as_admin(self):
...
+        nb_email_sent = len(mail.outbox) - outbox_initial_email_count
+
+        # 2.a and 2.b: proper template?
```
Comment (RignonNoel) [inline api_volontaria/apps/volunteer/tests/test_view_participations.py]: Idem

File: api_volontaria/settings.py [hunk 8]
```diff
@@ -173,11 +173,17 @@
...
+# Force local storage for unittests. Temporary.
+# if len(sys.argv) > 1 and sys.argv[1] == 'test':
+#     STATIC_URL = '/static/'
+#     STATICFILES_STORAGE = 'django.contrib.staticfiles.storage.StaticFilesStorage'
+# else:
+#     STATIC_URL = config('STATIC_URL', default='/static/')
+#     STATICFILES_STORAGE = config('STATICFILES_STORAGE',
+#                                  default='django.contrib.staticfiles.storage.StaticFilesStorage')
+# if STATICFILES_STORAGE == 'django.contrib.staticfiles.storage.StaticFilesStorage':
+#     STATIC_ROOT = 'static/'
```
Comment (RignonNoel) [inline api_volontaria/settings.py]: Code should not be commented, if needed you can remove it
PS: The old comment was cool to have in the settings in think

File: api_volontaria/settings.py [hunk 9]
```diff
@@ -266,3 +274,27 @@
...
+#     'django.contrib.staticfiles.finders.FileSystemFinder',
+#     'django.contrib.staticfiles.finders.AppDirectoriesFinder',
+# ]
+
+
+# STATICFILES_FINDERS = (
+#     'django.contrib.staticfiles.finders.FileSystemFinder',
+#     'django.contrib.staticfiles.finders.AppDirectoriesFinder',
+#     'django.contrib.staticfiles.finders.DefaultStorageFinder',
+#     )
```
Comment (RignonNoel) [inline api_volontaria/settings.py]: Remove commented code

**Review by RignonNoel — CHANGES_REQUESTED** [review]
Think to add all the generated static to the `.gitignore`

**Pull request merged**